scripts/crctool.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python3
+import sys
+import struct
+import argparse
+from zlib import crc32
+from pathlib import Path
+
+
+def bit_rev(val):
+    "Reverse bits of 32-bit words, stored as ints"
+    if isinstance(val, int):
+        return int('{:032b}'.format(val)[::-1], 2)
+    else:
+        output = []
+        for x in val:
+            output.append(int('{:032b}'.format(x)[::-1], 2))
+        return output
+
+
+def calculate_stm32_crc(data):
+    """Generate a CRC compatible with the STM32 hardware CRC
+
+        STM's CRC uses a normal CRC32 (MSB first) with a reset value of 0xFFFFFFFF
+        zlib uses a reversed CRC32 (LSB first) with a reset value of 0
+        To convert from zlib to STM formats we have to:
+        - bit reverse every 32-bit word of the input
+        - bit reverse the CRC generated by zlib.crc32
+        - bit invert the result as an unsigned number
+        from https://community.st.com/s/question/0D50X00009XkbXeSAJ/crc32-calculation-mismatch (post 3)
+    """
+    # TODO test if data len is a multiple of 4
```

Comment:
Looks like this TODO is already resolved.

Reply:
Yeah, I've removed it.